### PR TITLE
FormChooseCommit & Rebase onto: Display only potential commits

### PR DESCRIPTION
**Pull request by pmiossec**
so that choice is not overwhelming
and user has less data to process to find commit he wants

### Before

All the git history is displayed and user has to search for the commit to select:
image

### After

Only the commit that could be rebased are displayed and it's easier for user to find the good low bound one of commits to rebase:
image

## Test methodology 

- Manual

## Test environment(s) 

- Git Extensions 33.33.33
- Build 754b034fbe97b7e7973e5b43bfed7fc87cff1a3e
- Git 2.44.0.windows.1
- Microsoft Windows NT 10.0.22631.0
- .NET 8.0.1
- DPI 96dpi (no scaling)
- Portable: False
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 8.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/UserControls/RevisionGrid/FilterInfo.cs [hunk 1]
```diff
@@ -483,6 +488,11 @@ private void GetBranchRevisionFilter(ArgumentBuilder filter, Lazy<ObjectId?> cur
                 }
             }
 
+            if (!string.IsNullOrWhiteSpace(LastRevisionToDisplayHash))
+            {
+                filter.Add("..." + LastRevisionToDisplayHash);
```
Comment (mstv) [inline GitUI/UserControls/RevisionGrid/FilterInfo.cs:493]:
```suggestion
                filter.Add($"...{LastRevisionToDisplayHash}");
```

**Review by mstv — COMMENTED** [review]
+1, just a nit, have not run

File: GitUI/HelperDialogs/FormChooseCommit.cs [hunk 2]
```diff
@@ -12,11 +12,16 @@ private FormChooseCommit(GitUICommands commands)
             InitializeComplete();
         }
 
-        public FormChooseCommit(GitUICommands commands, string? preselectCommit, bool showArtificial = false, bool showCurrentBranchOnly = false)
+        public FormChooseCommit(GitUICommands commands, string? preselectCommit, bool showArtificial = false, bool showCurrentBranchOnly = false, string? lastRevisionToDisplayHash = null)
             : this(commands)
         {
             revisionGrid.MultiSelect = false;
             revisionGrid.ShowUncommittedChangesIfPossible = showArtificial;
+            if (lastRevisionToDisplayHash != null)
```
Comment (RussKie) [inline GitUI/HelperDialogs/FormChooseCommit.cs:20]:
```suggestion
            if (lastRevisionToDisplayHash is not null)
```

File: GitUI/UserControls/RevisionGrid/FilterInfo.cs [hunk 3]
```diff
@@ -194,6 +194,11 @@ public bool HasRevisionFilter
                 || ShowSimplifyByDecoration;
         }
 
+        /// <summary>
+        /// Hash of the last revision to display
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/FilterInfo.cs:198]:
```suggestion
        /// The hash of the last revision to display.
```

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 4]
```diff
@@ -1952,6 +1952,11 @@ public void ShowCurrentBranchOnly()
             PerformRefreshRevisions();
         }
 
+        public string LastRevisionToDisplayHash
+        {
+            set => _filterInfo.LastRevisionToDisplayHash = value;
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:1957]: ❗  No setter-only properties

File: GitUI/UserControls/RevisionGrid/FilterInfo.cs [hunk 5]
```diff
@@ -194,6 +194,11 @@ public bool HasRevisionFilter
                 || ShowSimplifyByDecoration;
         }
 
+        /// <summary>
+        /// The hash of the last revision to display.
+        /// </summary>
+        public string LastRevisionToDisplayHash { get; internal set; }
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/FilterInfo.cs:200]: What does "the last revision to display" mean?

**Review by RussKie — APPROVED** [review]

File: GitUI/UserControls/RevisionGrid/FilterInfo.cs (hunk 5 above)
Comment (pmiossec) [reply GitUI/UserControls/RevisionGrid/FilterInfo.cs:200]: This is the oldest commit (so the last one at the bottom) that will be displayed. So we use the commit hash to query history only until this commit.

Comment (RussKie) [reply GitUI/UserControls/RevisionGrid/FilterInfo.cs:200]: Please add this to the xml-doc.

**Comment by pmiossec** [comment]
I think it's good to go... merging?

**Pull request merged**